### ISSUE TITLE
Fredholm1 and Fredholm2 operators

### ISSUE THREAD

**Issue by mrava87** [PyLops/pylops#31]
Create generic operators in ``signalprocessing`` submodule performing Fredholm integrals of first and second kind as in https://en.wikipedia.org/wiki/Fredholm_integral_equation for 3 dimensional operators (note that when the kernel is 2d this is just  ``MatrixMult`` 1st kind and  ``Identity - MatrixMult``  2nd kind).

Such operators have practically being made within ``MDC`` and ``Marchenko`` but having something more generic will allow to build ``MDC`` as a chain of ``FFT.H Fredholm1 FFT`` and ``Marchenko`` as a ``Fredholm2`` with user defined kernel. 

**Main advantage**: when focusing on speed up this will be limited to the actual integral computation, which is nothing more than a dense matrix-vector or matrix-matrix operation. We can use using anything available open-source for Machine Learning (as fast matrix-matrix is all a neural network needs to be fast). Options:

* pycuda
* scikit-cuda
* tensorflow
* pytorch
* ...

**Comment by mrava87** [comment]
``Fredholm1`` implemented.

**Comment by mrava87** [comment]
``Fredholm2`` does not make much sense to implement as it can be easily made by combining primitive operators: ``Identity+Fredholm1``
